File: Structural-Patterns/Bridge/Java/Implementation/CardGateway.java
package Implementation;

public class CardGateway implements IPaymentGateway {
    @Override
    public void processPayment(double amount) {
        System.out.println("Processing payment of Rs." + amount + " via UPI.");
    }

    @Override
    public void refundPayment(String transactionId, double amount) {
        System.out.println("Refunding Rs." + amount + " for transaction " + transactionId + " via UPI.");
    }
}

File: Structural-Patterns/Bridge/C#/Bridge/Implementation/CardGateway.cs
public class CardGateway : IPaymentGateway {
    public void ProcessPayment(double amount) {
        Console.WriteLine("Processing payment of Rs." + amount + " via UPI.");
    }

    public void RefundPayment(String transactionId, double amount) {
        Console.WriteLine("Refunding Rs." + amount + " for transaction " + transactionId + " via UPI.");
    }
}
